### ISSUE TITLE
Why using weight normalization in StdConv2d?

### ISSUE THREAD

**Issue by xychenunc** [google-research/vision_transformer#76]
As stated in my question, is there any special reason to use weight normalization? Thank you!

def weight_standardize(w, axis, eps):
  """Subtracts mean and divides by standard deviation."""
  w = w - jnp.mean(w, axis=axis)
  w = w / (jnp.std(w, axis=axis) + eps)
  return w

class StdConv(nn.Conv):
  """Convolution with weight standardization."""

  def param(self, name, shape, initializer):
    param = super().param(name, shape, initializer)
    if name == 'kernel':
      param = weight_standardize(param, axis=[0, 1, 2], eps=1e-5)
    return param

**Comment by lucasb-eyer** [comment]
This is based on the BiT variant of ResNet from the [BigTransfer paper](https://arxiv.org/abs/1912.11370).
